se_apps/include/interface.h

```diff
@@ -109,526 +188,640 @@ static const float _focal_x = SceneK[0][0];
-		if (res != 1) {
-			std::cerr << "Invalid Raw file." << std::endl;
-
-		} else {
-			cameraOpen = true;
-			cameraActive = true;
-			_frame = -1;
-			_fps = fps;
-			_blocking_read = blocking_read;
-			fseeko(_pFile, 0, SEEK_SET);
-		}
-	};
-	ReaderType getType() {
-		return (READER_RAW);
-	}
-	inline bool readNextDepthFrame(uchar3* raw_rgb,
-			unsigned short int * depthMap) {
-
-		int total = 0;
-		int expected_size = 0;
-		unsigned int newImageSize[2];
-
-		get_next_frame();
-
-//		if (_frame < 2135)
-//		{
-//			_frame = 2135;
-//		}
-//
-//		std::cout << "Frame: " << _frame << std::endl;
+  private:
+    FILE* _pFile; /** Pointer to the open .raw file. */
+    uint2 _size; /** Dimensions of the input images. */
+
+  public:
+    /**
+     * Constructor using the ::ReaderConfiguration struct.
+     */
+    RawDepthReader(const ReaderConfiguration& config)
+      : DepthReader() {
+        // Copy configuration
+        _data_path = config.data_path;
+        _groundtruth_path = config.groundtruth_path;
+        _transform = config.transform;
+        // Open ground truth association file if supplied
+        if (_groundtruth_path != "") {
+          _gt_file.open(_groundtruth_path.c_str());
+          if(!_gt_file.is_open()) {
+            std::cout << "Failed to open ground truth association file "
+              << _groundtruth_path << std::endl;
+            _pFile = NULL;
+            return;
+          }
+          _pose_num = -1;
+        }
+        // Open raw file
+        std::string raw_filename = _data_path;
+        _pFile = fopen(raw_filename.c_str(), "rb");
+        size_t res = fread(&(_size), sizeof(_size), 1, _pFile);
+        cameraOpen = false;
+        cameraActive = false;
+        if (res != 1) {
+          std::cerr << "Invalid Raw file " << raw_filename << std::endl;
+          return;
+        } else {
+          cameraOpen = true;
+          cameraActive = true;
+          _frame = -1;
+          _fps = config.fps;
+          _blocking_read = config.blocking_read;
+          fseeko(_pFile, 0, SEEK_SET);
+        }
+      };
+
+    /**
+     * Old style constructor. Does not support ground truth loading.
+     *
+     * @deprecated Might be removed in the future.
+     */
+    RawDepthReader(std::string filename, int fps, bool blocking_read) :
+      DepthReader(), _pFile(fopen(filename.c_str(), "rb")) {
+
+        size_t res = fread(&(_size), sizeof(_size), 1, _pFile);
+        cameraOpen = false;
+        cameraActive = false;
+        if (res != 1) {
+          std::cerr << "Invalid Raw file." << std::endl;
+
+        } else {
+          cameraOpen = true;
+          cameraActive = true;
+          _frame = -1;
+          _fps = fps;
+          _blocking_read = blocking_read;
+          fseeko(_pFile, 0, SEEK_SET);
+        }
+      };
+
+    /**
+     * Get the type of the reader.
+     *
+     * \return ReaderType::READER_RAW
+     */
+    ReaderType getType() {
+      return (READER_RAW);
+    }
+
+    /**
```

Comment:
This comment does not really add any info to the function signature

Reply:
Removed in d0e0cf4dcde9e2be2904002f599c9a285638de8c.